common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -236,56 +236,27 @@ describe 'Problem', ->
         expect(@problem.el).toHaveHtml contents
         expect(window.SR.readTexts).toHaveBeenCalledWith ['no, try again']
 
-    it 'tests if all the capa buttons are disabled while submitting', (done)->
-      deferred = $.Deferred()
+    it 'tests if the submit button is disabled while submitting and the text changes on the button', ->
       self = this
-
-      runs = ->
-        spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
-          promise = undefined
-          callback
-            success: 'incorrect'
-            contents: 'Incorrect'
-          promise =
-            always: (callable) ->
-              callable()
-            done: (callable) ->
-              callable()
-        spyOn @problem, 'enableAllButtons'
-        @problem.submit()
-        expect(@problem.enableAllButtons).toHaveBeenCalledWith false, true
-        if jQuery.active == 0
-          deferred.resolve()
-        deferred.promise()
-
-      runs.call(self).then(->
-        expect(self.problem.enableAllButtons).toHaveBeenCalledWith true, true
-        return
-      ).always done
-
-    it 'tests the expected change in text of submit button', (done) ->
-      deferred = $.Deferred()
-      self = this
-
-      runs = ->
-        spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
-          promise = undefined
-          promise =
-            always: (callable) ->
-              callable()
-            done: (callable) ->
-              callable()
-        spyOn @problem.submitButtonLabel, 'text'
-        @problem.submit()
-        expect(@problem.submitButtonLabel.text).toHaveBeenCalledWith 'Submitting'
-        if jQuery.active == 0
-          deferred.resolve()
-        deferred.promise()
-
-      runs.call(self).then(->
-        expect(self.problem.submitButtonLabel.text).toHaveBeenCalledWith 'Submit'
-        return
-      ).always done
+      curr_html = @problem.el.html()
+      spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
+        # At this point enableButtons should have been called, making the submit button disabled with text 'submitting'
+        expect(self.problem.submitButton).toHaveAttr('disabled');
+        expect(self.problem.submitButtonLabel.text()).toBe('Submitting');
+        promise = undefined
```

Comment:
I was getting a linting error from my IDE, `unused local variable promise` if I didn't initialize it. Though I did just notice there's other tests I didn't touch that have this linting error that pass our own style checks. I'll go through and update these to make them consistent, then this should be good to go?